client/my-sites/backup/hooks.js

```diff
@@ -61,15 +61,9 @@ export const useFirstMatchingBackupAttempt = (
 	siteId,
 	{ before, after, successOnly, sortOrder } = {}
 ) => {
-	const dispatch = useDispatch();
-
-	useEffect( () => {
-		dispatch( requestRewindCapabilities( siteId ) );
-	}, [ siteId ] );
```

Comment:
@jsnajdr I initially tried to replace `requestRewindCapabilities` with `fetchSiteFeatures ` but it caused an infinite render loop when visiting https://container-elastic-elgamal.calypso.live/backup with a WoA site so I removed that in 98775b5f7f6f1dd745ff72301f6571a2b9d839d4. 

I saw [you recently worked in this area](https://github.com/Automattic/wp-calypso/pull/60756) and was curious if you could help me figure out if removing this would be okay or how I could rewrite it to avoid re-rendering?

Reply:
This hook (`useFirstMatchingBackupAttempt`) is used by the `RealtimeStatus` component and the fetch happens when that component mounts.

Now `RealtimeStatus` is rendered by `BackupStatus`, conditionally when site features have finished fetching. The code is like:
```js
function BackupStatus() {
  if ( isFetchingSiteFeatures ) {
    return <BackupPlaceholder />;
  }
  return <RealtimeStatus />;
}
```
And the infinite loop goes like this:
1. Someone (e.g. the top level `Layout` with `QuerySiteFeatures`) triggers a site features fetch.
2. `BackupStatus` renders a placeholder.
3. Site features fetch finishes.
4. `BackupStatus` renders `RealtimeStatus` instead of the placeholder.
5. That means that `RealtimeStatus` _mounts_ and on mount it triggers a site features fetch again.
6. `BackupStatus` renders the placeholder again because `isFetchingSiteFeatures` is true again.
7. Site features fetch finishes and `RealtimeStatus` is mounted again...

We need to be more careful with `fetchSiteFeatures` and `QuerySiteFeatures`. The features are always queried in the top-level `Layout` component, so no one else should need to query them explicitly.

Ideally, the fetch would happen only when i) we're not already fetching and ii) the fetch hasn't already finished. In other words, if site features are already in state, cache them forever and don't even re-fetch them.